caas/kubernetes/provider/k8s.go

```diff
@@ -183,10 +216,34 @@ func (k *kubernetesClient) validateOperatorStorage() (string, error) {
 	return storageClass, errors.Trace(err)
 }
 
-// PrepareForBootstrap prepares for bootstrapping a controller.
-func (k *kubernetesClient) PrepareForBootstrap(ctx environs.BootstrapContext) error {
-	_, err := k.validateOperatorStorage()
-	return err
+// PrepareForBootstrap prepares for bootstraping a controller.
+func (k *kubernetesClient) PrepareForBootstrap(ctx environs.BootstrapContext, controllerName string) error {
+	k.namespace = controllerName
+	alreadyExistErr := errors.AlreadyExistsf("namespace %q, choose another namespace name then try again", k.namespace)
```

Comment:
this will append "already exists" to the message
also, the user is not choosing a namespace name, they are choosing a controller name
you want something like:
errors.NewAlreadyExistsf(nil, "a controller called %q already exists on this k8s cluster.\nPlease bootstrap again and choose a different controller name.", k.namespace)

Reply:
changed